lib/database/points.php

```diff
@@ -0,0 +1,441 @@
+        $gameInfo = TrueRetroRatio_WorkAroundV1::getGameNameAndConsole($gameID);
+        if ($gameInfo == null) {
+            return null;
+        }
+
+        $bonus = mb_substr($gameInfo->Title, 0, 3) === "~Bo";
+
+        $achsNotGrouped = TrueRetroRatio_WorkAroundV1::getAchsForGame($gameID);
+        if ($achsNotGrouped == null) {
+            return null;
+        }
+        $achieved = TrueRetroRatio_WorkAroundV1::getAchieved($gameID, $HC);
+        if (empty($achieved)) {
+            return null;
+        }
+        //getAchieved doesn't return achievements with 0 players
+        //Don't know if it's possible to do with 1 sql request
+        foreach ($achsNotGrouped as $ach) {
+            $ach->Achieved = 0;
+            foreach ($achieved as $a) {
+                if ($ach->ID == $a->ID) {
+                    $ach->Achieved = (int) $a->Achieved;
+                    break;
+                }
+            }
+        }
+
+        $achsGrouped = TrueRetroRatio_WorkAroundV1::groupAchievements($achsNotGrouped, $bonus);
+
+        $realID = null;
+        if ($bonus) {
+            $realID = TrueRetroRatio_WorkAroundV1::getParentIdForBonus($gameInfo);
+            //returns null if not found
+        }
+        $gotParent = $realID != null;
+        if (!$gotParent) {
+            $realID = $gameID;
+        }
+
+        return TrueRetroRatio_WorkAroundV1::
+        calculatePlayersPerRevision($achsGrouped, $realID, $gotParent, $HC, $checkUnTrThr);
+    }
+}
+
+class TrueRetroRatio
+{
+    private function getAchievementsGroupedByRevision($gameID, $hardcore)
+    {
+        //Here will be a normal code with 1-2 requests in V2 instead of this
+
+        $checkUnTrackedThr = 500;
+        //for popular game with not checking them the score changes only by ~0.5% max but speedup a lot
+        //Game          SMW  DKC3   SM3   SoR  SotN  R-Type
+        //HardcPlayers 8300  5000  3500  2000  1000   500
+        //Check UnTr.   11s  4.6s  3.6s  1.9s  0.6s   0.3s
+        //!Check UnTr. 1.2s  0.9s  0.6s  0.4s  0.2s  <0.1s
+
+        return (new TrueRetroRatio_WorkAroundV1())->getAchievementsGroupedByRevision($gameID, $hardcore, $checkUnTrackedThr);
+    }
+
+    private function boostRevisions($achsGrouped)
+    {
+        //Because old players will start earn achievements immediately
+        if (count($achsGrouped) < 2) {
+            return $achsGrouped;
+        }
+        $minRevPlayers = (int) round(
+            ($achsGrouped[0]->players < 100)
+                ? 0.2 * $achsGrouped[0]->players
+                : 2 * sqrt($achsGrouped[0]->players)
+        );
+        for ($i = 1; $i < count($achsGrouped); $i++) {
+            $achsGrouped[$i]->players = $achsGrouped[$i]->players + $minRevPlayers;
+        }
+
+        return $achsGrouped;
+    }
+
+    private function getPercentageOfBoost($players)
+    {
+        $boost0At = 1000;
+        if ($players >= $boost0At) {
+            return 0;
+        }
+
+        //$boost50At = 200;
```

Comment:
If this is code that ended up becoming unused, it should be removed.

Reply:
It not as much as code, as explanation how the "magic number" few lines below appeared.